packages/dashboard/src/components/internalDashboard/gestures/index.spec.tsx

```diff
@@ -0,0 +1,180 @@
+import React from 'react';
+import { renderHook, act } from '@testing-library/react-hooks';
+import { Provider } from 'react-redux';
+
+import { configureDashboardStore } from '../../../store';
+import { RecursivePartial } from '../../../types';
+import { DashboardState } from '../../../store/state';
+import { MockDashboardFactory } from '../../../../testing/mocks';
+
+import { useGestures } from './index';
+import { anchorable, gestureable, idable } from './determineTargetGestures';
+
+const TestProvider: React.FC<{
+  storeArgs?: RecursivePartial<DashboardState>;
+}> = ({ storeArgs, children }) => <Provider store={configureDashboardStore(storeArgs)}>{children}</Provider>;
+
+const widgetGestureAttribute = gestureable('widget');
+const resizeGestureAttribute = gestureable('resize');
+const anchorAttribute = anchorable('bottom');
+const idAttribute = idable('widget-1');
+
+const mockWidget = document.createElement('div');
```

Comment:
would be preferable to just construct these via JSX rather than imperatively across all tests

Reply:
That is the way I initially wrote the test, but I was running into an issue where getAttribute which is important for determining the gesture was undefined.